server/Controllers/GameControllers/snake.js

```diff
@@ -1,12 +1,115 @@
+const { SNAKE } = require('./gameConstants');
+
 class Snake {
-  constructor(playerId, initialPositionX, initialPositionY,
-    initialDirection, snakeColor, initialSnakeLength) {
+  /**
+   * @param  {string} playerId
+   * @param  {Object} initialProperties initial values for snake body and motion.
+   * @param  {Object} game instance of game class for status of current game. 
+   */
+  constructor(playerId, initialProperties, game) {
     this.ownerId = playerId;
-    this.head = { x: initialPositionX, y: initialPositionY };
-    this.direction = initialDirection;
-    this.color = snakeColor;
-    this.bodyLength = initialSnakeLength;
+    this.head = { x: initialProperties.position.x, y: initialProperties.position.y };
+    this.direction = initialProperties.direction;
+    this.velocity = initialProperties.velocity;
+    this.color = initialProperties.color;
+    this.length = initialProperties.length;
+    this.eatFood = game.eatFood;
```

Comment:
just store `this.game = game` and then use it everywhere.

Reply:
But, that will affect a little bit of performance, there is a 20ms timeout to send the next direction to the clients. And by doing this I am actually doing 4(2 for snake one and 2 for snake two) memory reference unnecessary for each move of a snake.